### PR TITLE
Add aggregation bitfield overlap check 

### PR DESCRIPTION
**Pull request by terencechain**
This PR modifies `AggregateAttestation` to use `Overlap` instead of `Contains`, and adds two tests

### CODE REVIEW

File: beacon-chain/blockchain/forkchoice/process_attestation.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func (s *Store) aggregateAttestation(ctx context.Context, att *ethpb.Attestation
 	if a, ok := s.attsQueue[root]; ok {
 		a, err := helpers.AggregateAttestation(a, att)
 		if err != nil {
-			return nil
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/forkchoice/process_attestation.go]: Can you add `traceutil.AnnotateError(span, err)` please?

**Comment by terencechain** [comment]
Addressed via #3761

**Comment by prestonvanloon** [comment]
More conflicts. Please resolve

**Comment by terencechain** [comment]
Forgot to this close. This is addressed in #3804

**Pull request closed without merging**